### ISSUE TITLE
Data type changes in view's referenced tables breaks information schema

### ISSUE THREAD

**Issue by proddata** [crate/crate#14377]
### CrateDB version

5.3.1

### CrateDB setup information

local single node

### Problem description

When altering the data type in view's referenced tables queries of the  information schema return an exception, e.g..:

```
SQLParseException[Cannot cast expressions from type `timestamp without time zone_array` to type `timestamp without time zone`]
```

<details><summary>Stacktrace</summary>
<p>
io.crate.exceptions.SQLParseException: Cannot cast expressions from type `timestamp without time zone_array` to type `timestamp without time zone`
	at io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:162)
	at io.crate.exceptions.SQLExceptions.prepareForClientTransmission(SQLExceptions.java:151)
	at io.crate.rest.action.SqlHttpHandler.sendResponse(SqlHttpHandler.java:163)
	at io.crate.rest.action.SqlHttpHandler.lambda$channelRead0$0(SqlHttpHandler.java:119)
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:863)
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:841)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2194)
	at io.crate.rest.action.RestResultSetReceiver.fail(RestResultSetReceiver.java:88)
	at io.crate.protocols.postgres.RetryOnFailureResultReceiver.fail(RetryOnFailureResultReceiver.java:115)
	at io.crate.action.sql.RowConsumerToResultReceiver.consumeIt(RowConsumerToResultReceiver.java:114)
	at io.crate.action.sql.RowConsumerToResultReceiver.accept(RowConsumerToResultReceiver.java:57)
	at io.crate.execution.engine.InterceptingRowConsumer.lambda$tryForwardResult$1(InterceptingRowConsumer.java:92)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1589)
Caused by: io.crate.exceptions.ConversionException: Cannot cast expressions from type `timestamp without time zone_array` to type `timestamp without time zone`
	at io.crate.expression.scalar.cast.CastFunctionResolver.generateCastFunction(CastFunctionResolver.java:64)
	at io.crate.expression.symbol.Symbol.cast(Symbol.java:76)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.visitCast(ExpressionAnalyzer.java:565)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.visitCast(ExpressionAnalyzer.java:443)
	at io.crate.sql.tree.Cast.accept(Cast.java:45)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.visitArithmeticExpression(ExpressionAnalyzer.java:851)
	at io.crate.analyze.expressions.ExpressionAnalyzer$InnerExpressionAnalyzer.visitArithmeticExpression(ExpressionAnalyzer.java:443)
	at io.crate.sql.tree.ArithmeticExpression.accept(ArithmeticExpression.java:67)
	at io.crate.analyze.expressions.ExpressionAnalyzer.convert(ExpressionAnalyzer.java:244)
	at io.crate.analyze.relations.select.SelectAnalysis.toSymbol(SelectAnalysis.java:63)
	at io.crate.analyze.relations.select.SelectAnalyzer$InnerVisitor.visitSingleColumn(SelectAnalyzer.java:63)
	at io.crate.analyze.relations.select.SelectAnalyzer$InnerVisitor.visitSingleColumn(SelectAnalyzer.java:59)
	at io.crate.sql.tree.SingleColumn.accept(SingleColumn.java:78)
	at io.crate.analyze.relations.select.SelectAnalyzer.lambda$analyzeSelectItems$0(SelectAnalyzer.java:54)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1092)
	at io.crate.analyze.relations.select.SelectAnalyzer.analyzeSelectItems(SelectAnalyzer.java:54)
	at io.crate.analyze.relations.RelationAnalyzer.visitQuerySpecification(RelationAnalyzer.java:440)
	at io.crate.analyze.relations.RelationAnalyzer.visitQuerySpecification(RelationAnalyzer.java:115)
	at io.crate.sql.tree.QuerySpecification.accept(QuerySpecification.java:101)
	at io.crate.analyze.relations.RelationAnalyzer.visitQuery(RelationAnalyzer.java:150)
	at io.crate.analyze.relations.RelationAnalyzer.visitQuery(RelationAnalyzer.java:115)
	at io.crate.sql.tree.Query.accept(Query.java:72)
	at io.crate.analyze.relations.RelationAnalyzer.analyze(RelationAnalyzer.java:133)
	at io.crate.analyze.relations.RelationAnalyzer.analyze(RelationAnalyzer.java:139)
	at io.crate.metadata.view.ViewInfoFactory.create(ViewInfoFactory.java:66)
	at io.crate.metadata.doc.DocSchemaInfo.getViewInfo(DocSchemaInfo.java:193)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.HashMap$KeySpliterator.tryAdvance(HashMap.java:1736)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:161)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1931)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:280)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1932)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:161)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
	at java.base/java.util.stream.Streams$ConcatSpliterator.tryAdvance(Streams.java:723)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1931)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at java.base/java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1931)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.lambda$initPartialTraversalState$0(StreamSpliterators.java:292)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.fillBuffer(StreamSpliterators.java:206)
	at java.base/java.util.stream.StreamSpliterators$AbstractWrappingSpliterator.doAdvance(StreamSpliterators.java:169)
	at java.base/java.util.stream.StreamSpliterators$WrappingSpliterator.tryAdvance(StreamSpliterators.java:298)
	at java.base/java.util.Spliterators$1Adapter.hasNext(Spliterators.java:681)
	at io.crate.data.CollectingBatchIterator.moveNext(CollectingBatchIterator.java:129)
	at io.crate.data.MappedForwardingBatchIterator.moveNext(MappedForwardingBatchIterator.java:41)
	at io.crate.data.LimitingBatchIterator.moveNext(LimitingBatchIterator.java:57)
	at io.crate.action.sql.RowConsumerToResultReceiver.consumeIt(RowConsumerToResultReceiver.java:75)
	... 5 more

<p>
</details> 

### Steps to Reproduce

```sql
CREATE TABLE t01 (ts TIMESTAMP);
CREATE VIEW v01 AS SELECT ts::TIMESTAMP - '1 hour'::INTERVAL FROM t01;
DROP TABLE t01;
CREATE TABLE t01 (ts ARRAY(TIMESTAMP));
SELECT * FROM information_schema.tables
```

### Actual Result

information schema tables are largely broken

### Expected Result

Information schema tables are still available.
The view is broken.

**Comment by matriv** [comment]
Is the issue fixed if the table is:
- recreated again with original schema
- the view is dropped and recreated (on top of the new schema)
?

**Comment by proddata** [comment]
It is resolved by ...
- dropping the view
or
- dropping the table

you can also recreate the table with a different schema again, as long as the datatype is compatible

**Comment by BaurzhanSakhariev** [comment]
Thanks for reporting, fixed in https://github.com/crate/crate/pull/14444.

We also added` /* Corrupted view, needs fix */` to the broken views statements 
in `pg_catalog.pg_views.definition` and in `information_schema.views.view_definition` columns.

**Comment by BaurzhanSakhariev** [comment]
Re-opening due to https://github.com/crate/crate/issues/14382

We added "corrupted view, needs fix" for all broken views. 

However, due to the fact, that views are stored without schema, users might see that "corrupted" for normal views if table was created under some custom schema. 

For now we are adding the comment only for views having anything but `ResourceUnknownException`, ie. adding comment only for broken views but not for "abandoned" ones.

Once we fix  https://github.com/crate/crate/issues/14382 we will re-iterate on this issue and add comment on "abandoned" views as well.

**Comment by mfussenegger** [comment]
> Information schema is working now. Nevertheless, re-opening due to #14382
> 
> We added "corrupted view, needs fix" hint for all broken views. Due to the fact, that views are stored without schema, users might see that "corrupted" hint for normal views if view was created under some custom schema.
> 
> For now we are adding the comment only for views having anything but `ResourceUnknownException`, ie. adding comment only for broken views but not for "abandoned" ones.
> 
> Once we fix #14382 we will re-iterate on this issue and add comment on "abandoned" views as well.

Done with https://github.com/crate/crate/pull/14494
